crates/infra/cli/src/toolchains/napi/resolver.rs

```diff
@@ -9,47 +9,56 @@ pub enum NapiPackageKind {
     Platform(String),
 }
 
-pub struct NapiResolver;
+pub struct NapiResolver {
+    name: &'static str,
```

Comment:
I suggest using an `enum` for the name here (`NapiResolver::new(NapiPackageId::Solidity)`), or exposing two constructors instead (`NapiResolver::solidity()` and `NapiResolver::test_lang()`).

- The path will be stored in a single place (here in the resolver) to make it easier to maintain the current infra scripts.
- This will make sure that callsite is type checked, which also makes it easier to refactor out and make these build scripts generic later.
- It lets us use complete package names in string literals, instead of `format!()`, which allows us to do string search/replace over the whole codebase.

Reply:
The reason I use a string value is to make external configuration possible, rather than encoding the package structure in the type system. However I acknowledge that parametricity is pointless in this case.

I'll add constructors.

Reply:
Sounds great. Thank you! In that case, and for the third reason above, I suggest storing the full package names. I realize the existing code indicates that they should share the same prefix, but using full string literals for package/folder identifiers is really important to keep infra maintainable:

```rust
    pub fn testlang() -> Self {
        Self {
            rust_crate: "testlang_npm_crate",
            typescript_crate: "testlang_npm_package",
        }
    }

    pub fn solidity() -> Self {
        Self {
            rust_crate: "solidity_npm_crate",
            typescript_crate: "solidity_npm_package",
        }
    }
```

Reply:
Done